docs/content/en/docs/yaml-crd-ref/taskdefinition.md

```diff
@@ -3,3 +3,258 @@ title: KeptnTaskDefinition
+     ```
+
+     See
+     [Parameterized functions](../implementing/tasks/#parameterized-functions)
+     for more information.
+
+  * **secureParameters** -- An optional field used to pass a Kubernetes secret.
+    The `secret` value is the Kubernetes secret name
+    that is mounted into the runtime and made available to functions
+    using the `SECURE_DATA` environment variable.
+    For example:
+
+    ```yaml
+    secureParameters:
+      secret: slack-token
+    ```
+
+    Note that, currently, only one secret can be passed.
+
+    See [Create secret text](../implementing/tasks/#create-secret-text)
+    for details.
+
+## Usage
+
+A Task executes the TaskDefinition of a
+[KeptnApp](app.md) or
+[workload](https://kubernetes.io/docs/concepts/workloads/).
+The execution is done by spawning a Kubernetes
+[Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/)
+to handle a single Task.
+In its state, it tracks the current status of this Kubernetes Job.
+
+The `function` is coded in JavaScript
+and executed in
+[Deno](https://deno.com/runtime),
+which is a lightweight runtime environment
+that executes in your namespace.
+Note that Deno has tighter restrictions
+for permissions and importing data
+so a script that works properly elsewhere
+may not function properly when run in Deno.
+
+A task can be executed either pre-deployment or post-deployment
+as specified in the `Deployment` resource;
+see
+[Pre- and post-deployment tasks](../implementing/integrate/#pre--and-post-deployment-checks)
+for details.
+Note that the annotation identifies the task by `name`.
+This means that you can modify the `function` code in the resource definition
+and the revised code is picked up without additional changes.
+
+## Examples
+
+### Example 1: inline script
+
+This example defines a full-fledged Deno script
+within the `KeptnTaskDefinition` YAML file:
+
+```yaml
+apiVersion: lifecycle.keptn.sh/v1alpha3
+kind: KeptnTaskDefinition
+metadata:
+  name: hello-keptn-inline
+spec:
+  function:
+    inline:
+      code: |
+        let text = Deno.env.get("DATA");
+        let data;
+        let name;
+        data = JSON.parse(text);
+
+        name = data.name
+        console.log("Hello, " + name + " new");
+```
+
+### Example 2: httpRef script
+
+This example fetches the Deno script from a remote webserver at runtime:
+
+```yaml
+apiVersion: lifecycle.keptn.sh/v1alpha3
+kind: KeptnTaskDefinition
+metadata:
+  name: hello-keptn-http
+spec:
+  function:
+    httpRef:
+      url: "https://www.example.com/yourscript.js"
+```
+
+For another example, see the
+[sample-app](https://github.com/keptn-sandbox/lifecycle-toolkit-examples/blob/main/sample-app/version-1/app-pre-deploy.yaml).
+
+See the
+[sample-app/version-1](https://github.com/keptn-sandbox/lifecycle-toolkit-examples/blob/main/sample-app/version-1/app-pre-deploy.yaml)
+PodtatoHead example for a more complete example.
+
+### Example 3: functionRef
```

Comment:
To the examples, I would maybe also add an example about referencing a configmap, which contains the function code

Reply:
Do you know where I can find such an example?